### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by maelk**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1,88 +1,200 @@
...
+                  192.168.111.249
+              }
+          }
 ```
 
-## hostSelector Examples
+## Machine
+
+A Machine is a Cluster API core object representing a Kubernetes node. A machine
+refers to a KubeadmConfig and to a BaremetalMachine
```
Comment (Xenwar) [inline docs/api.md]: A machine has a reference to a KubeadmConfig and a BaremetalMachine

File: docs/deployment_workflow.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  the KubeadmConfig :
+
+```yaml
+nodeRegistration:
+  name: '{{ ds.meta_data.name }}'
+  kubeletExtraArgs:
+    node-labels: 'metal3.io/uuid={{ ds.meta_data.uuid }}'
+```
+
+## Deploying the CRS
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: CRs

File: docs/deployment_workflow.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+You can deploy the CRS all at the same time, the controllers will take care of
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: CRs

File: docs/deployment_workflow.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+following the correct flow.
+An outline of the workflow is below.
+
+1. The CAPI controller will set the OwnerRef on the BaremetalCluster referenced
+   by the Cluster
+1. The CAPBM controller will verify the apiEndpoint field and populate the
+   status with ready set to true and apiEndpoint to the content of apiEndpoint.
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: ready field

File: docs/deployment_workflow.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+1. Once the machine has userdata, OwnerRef and bootstrapReady properly set, the
+   CAPBM controller will select, if possible, a BareMetalHost that matches the
+   criteria, or wait until one is available. If matched, the CAPBM controller
+   will create a secret with the userData. and set the BareMetalHost spec
+   accordingly to the BareMetalMachine specs.
+1. The BareMetal Operator will then start the deployment.
+1. After deployment, the BaremetalHost will be in provisioned state. However,
+   initialization is not complete. If deploying without cloud provider, CAPBM
+   will wait until the target cluster is up and the node appears. It will fetch
+   the node by matching the label "metal3.io/uuid=bmh-uuid". It will set the
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: if `bmh-uuid` is not static, indicate it by making it a variable.

File: docs/dev-setup.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+a dummy `BareMetalHost` object.  There’s no requirement to actually run the
+`baremetal-operator` to test the reconciliation logic of the actuator.
+
+Refer to the [baremetal-operator developer
+documentation](https://github.com/metal3-io/baremetal-operator/blob/master/docs/dev-setup.md)
+for instructions and tools for creating BareMetalHost objects.
+
+### Deploy CAPI and CAPBM
+
+The following command will deploy the controllers fro CAPI, CABPK and CAPBM and
```
Comment (Xenwar) [inline docs/dev-setup.md]: from

File: docs/api.md [hunk 7]
```diff
@@ -1,88 +1,200 @@
...
+
+The BareMetalMachine contains information related to the deployment of the
+BareMetalHost such as the image and the host selector. For each machine, there
+must be a BareMetalMachine.
+
+The fields are :
+
+* **image** -- This includes two sub-fields, `url` and `checksum`, which
+  include the URL to the image and the URL to a checksum for that image.  These
+  fields are required.  The image will be used for provisioning of the
```
Comment (Jaakko-Os) [inline docs/api.md]: There are a few occurrences in the text with two spaces after '.'
Please have only one space after '.'

File: docs/deployment_workflow.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    node-labels: 'metal3.io/uuid={{ ds.meta_data.uuid }}'
+```
+
+## Deploying the CRs
+
+You can deploy the CRs all at the same time, the controllers will take care of
+following the correct flow.
+An outline of the workflow is below.
+
+1. The CAPI controller will set the OwnerRef on the BaremetalCluster referenced
```
Comment (Jaakko-Os) [inline docs/deployment_workflow.md]: I suppose a running number list is what is wanted here? Now its all '1.'
Maybe a bulleted list is sufficient.

Comment (maelk) [reply docs/deployment_workflow.md]: That's how it works. the output will be a numbered list

**Pull request merged**